test/src/graph/multiReturnTest.cpp

```diff
@@ -1,40 +1,330 @@
 	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytrace, nullptr));
-	EXPECT_RGL_SUCCESS(rgl_node_points_compact_by_field(&compact, RGL_FIELD_IS_HIT_I32));
-	EXPECT_RGL_SUCCESS(rgl_node_points_yield(&yield, mrFields.data(), mrFields.size()));
+	EXPECT_RGL_SUCCESS(rgl_node_points_format(&format, fields.data(), fields.size()));
+	EXPECT_RGL_SUCCESS(rgl_node_points_ros2_publish(&publish, "MRTest_Lidar_Without_MR", "world"));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(rays, transform));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(transform, raytrace));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(raytrace, format));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(format, publish));
+
+	// Camera
+	const rgl_mat3x4f cameraPose = Mat3x4f::TRS(Vec3f{-8.0f, 1.0f, 5.0f}, {90.0f, 30.0f, -70.0f}).toRGL();
+	constructCameraGraph(cameraPose);
+
+	int frameId = 0;
+	while (true) {
+
+		const auto newPose = (entitiesTransforms.at(0) *
+		                      Mat3x4f::translation(0.0f, std::abs(std::sin(frameId * 0.05f)) * gapRange.y(), 0.0f))
+		                         .toRGL();
+		EXPECT_RGL_SUCCESS(rgl_entity_set_pose(entities.at(0), &newPose));
+
+		ASSERT_RGL_SUCCESS(rgl_graph_run(cameraRays));
+		ASSERT_RGL_SUCCESS(rgl_graph_run(rays));
+		ASSERT_RGL_SUCCESS(rgl_graph_run(mrRays));
+
+		std::this_thread::sleep_for(50ms);
+		frameId += 1;
+	}
+}
+#endif
+
+#if RGL_BUILD_ROS2_EXTENSION
+/**
+ * This test verifies how changing the beam divergence affects the results obtained with the multi return feature enabled.
+ * Three cubes arranged in the shape of a stairs are placed on the scene. LiDAR aims the only ray at the center of the middle “stair",
+ * during the test the beam divergence angle is increased/decreased periodically.
+ */
+TEST_F(GraphMultiReturn, stairs)
+{
+	/*
+	 *                    ____        ^
+	 *               ____|            | Z+
+	 *          ____| middle step     |
+	 *         |                       ----> X+
+	 */
+
+	GTEST_SKIP(); // Comment to run the test
+
+	// Scene
+	const float stairsBaseHeight = 0.0f;
+	const float stepWidth = 1.0f;
+	const float stepHeight = vlp16LidarHBeamDiameter + 0.1f;
+	const float stepDepth = 0.8f;
+	const Vec3f stairsTranslation{2.0f, 0.0f, 0.0f};
+
+	spawnStairsOnScene(stepWidth, stepHeight, stepDepth, stairsBaseHeight, stairsTranslation);
+
+	// Lidar
+	const std::vector<rgl_mat3x4f> raysTf{Mat3x4f::TRS({0.0f, 0.0f, 0.0f}, {90.0f, 0.0f, 90.0f}).toRGL()};
+
+	const float lidarMiddleStepDist = 1.5f * vlp16LidarObjectDistance;
+	const Vec3f lidarTransl{-lidarMiddleStepDist + stepDepth, 0.0f, stepHeight * 1.5f};
+
+	const rgl_mat3x4f lidarPose{(Mat3x4f::translation(lidarTransl + stairsTranslation)).toRGL()};
+
+	// Lidar with MR
+	const float beamDivAngle = vlp16LidarHBeamDivergence;
+	constructMRGraph(raysTf, lidarPose, beamDivAngle, true);
+
+	// Camera
+	rgl_mat3x4f cameraPose = Mat3x4f::translation({0.0f, -1.5f, stepHeight * 3 + 1.0f}).toRGL();
+	constructCameraGraph(cameraPose);
```

Comment:
Camera graph does not seem to be run?

Reply:
[Here](https://github.com/RobotecAI/RobotecGPULidar/pull/290/files#diff-536edcfe4b5e3529fc9e502b5fbb93a51c7f50cf1f2a5d623ec6f336c62720e1R291)

Reply:
Ok, that's clear, thank you for explanation!